### PR TITLE
Y24-249 - Fixed results not being returned properly

### LINKED ISSUE #1881: Y24-249 - [BUG] Filtering results for PacBio runs freezes page
**Describe the bug**
When filtering results for a PacBio run, the results will not load and refreshing will not resolve this.

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to the PacBio runs page
2. Input anything into the search box (including valid run names)
3. Search against run names
4. You will notice nothing happens and there will be an error in the console (see screenshot for info)
5. Refreshing the page will cause infinite loading until navigating elsewhere in Traction

**Expected behaviour**
The runs should be filtered and returned properly like they do in other parts of the UI (like pools).

**Screenshots**
Screenshot 2024-08-14 at 11 33 45
Screenshot 2024-08-14 at 11 34 12

### PR DESCRIPTION
**Pull request by SHIV5T3R**
Closes #1881 

#### Changes proposed in this pull request

- Ensures filterOptions is an array before filtering

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

**Review by seenanair — APPROVED** [review]

File: src/composables/useQueryParams.js [hunk 1]
```diff
@@ -42,6 +42,7 @@ export default function useQueryParams() {
     }
     let searchValue = filter_input.value
     if (
+      Array.isArray(filterOptions) &&
```
Comment (BenTopping) [inline src/composables/useQueryParams.js:45]: If filterOptions is not an array what is it for the runs page? I think it should always be passed in as an array.

Comment (SHIV5T3R) [reply src/composables/useQueryParams.js:45]: It's slightly different, not sure why tho. Have a look at `src/views/pacbio/PacbioPoolIndex.vue` compared to `src/views/pacbio/PacbioRunIndex.vue`

Comment (BenTopping) [reply src/composables/useQueryParams.js:45]: Ah. I dont think it needs to be a ref in PacbioRunIndex as its not dynamic so if you remove that does it work?

Comment (SHIV5T3R) [reply src/composables/useQueryParams.js:45]: seems to break the runs page. not sure if it's worth refactoring. what do you think?

Comment (BenTopping) [reply src/composables/useQueryParams.js:45]: Sorry I meant remove the ref function wrapper but keep the filterOptions as a normal array.

**Review by BenTopping — APPROVED** [review]

**Pull request merged**